lib/src/framework.dart

```diff
@@ -281,6 +281,12 @@ abstract class HookState<R, T extends Hook<R>> {
       .._isOptionalRebuild = false
       ..markNeedsBuild();
   }
+
+  /// The [diagnosticsProperty] is the default propertity to add to `properties`
+  /// param of the [Element.debugFillProperties].
+  /// If it's null, it will be skipped
+  @visibleForTesting
+  DiagnosticsProperty get diagnosticsProperty => null;
```

Comment:
I think it would be more logical to use same pattern as everywhere else in Flutter: `void debugFillProperties(properties)`

Reply:
It would probably make sense to have `HookState` implement `Diagnosticable`